src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/CropEdgeGeometry.cpp

```diff
@@ -0,0 +1,410 @@
+  destEdgesAttrMatrix.resizeTuples({numEdges});
+
+  std::vector<bool> edgesMask(numEdges, false);
+  std::vector<bool> vertexReferenced(numVertices, false);
+  std::unordered_map<uint64, std::tuple<float32, float32, float32>> interpolatedValuesMap;
+
+  for(usize i = 0; i < numEdges; ++i)
+  {
+    uint64 v0 = srcEdges[2 * i + 0];
+    uint64 v1 = srcEdges[2 * i + 1];
+
+    // Validate vertex indices
+    if(v0 >= numVertices)
+    {
+      return MakeErrorResult(to_underlying(ErrorCodes::InvalidVertexIndex), fmt::format("Edge at index {} with value {} references an invalid vertex index.", 2 * i, v0));
+    }
+
+    if(v1 >= numVertices)
+    {
+      return MakeErrorResult(to_underlying(ErrorCodes::InvalidVertexIndex), fmt::format("Edge at index {} with value {} references an invalid vertex index.", 2 * i + 1, v1));
+    }
+
+    // Determine if each vertex is inside or outside
+    float32 x1 = srcVertices[3 * v0 + 0];
+    float32 y1 = srcVertices[3 * v0 + 1];
+    float32 z1 = srcVertices[3 * v0 + 2];
+    bool v0_inside = is_inside(x1, y1, z1, boundingBox);
+
+    float32 x2 = srcVertices[3 * v1 + 0];
+    float32 y2 = srcVertices[3 * v1 + 1];
+    float32 z2 = srcVertices[3 * v1 + 2];
+    bool v1_inside = is_inside(x2, y2, z2, boundingBox);
+
+    bool edgeOutsideBoundary = !v0_inside && !v1_inside;
+    bool edgeInsideBoundary = v0_inside && v1_inside;
+    bool edgeIntersectingBoundary = (v0_inside && !v1_inside) || (!v0_inside && v1_inside);
+
+    if(edgeOutsideBoundary)
+    {
+      continue;
+    }
+
+    if(edgeIntersectingBoundary)
+    {
+      // Found an edge intersecting the boundary
+      if(behavior == BoundaryIntersectionBehavior::FilterError)
+      {
+        // Throw a filter error
+        float32 x_inside = v0_inside ? srcVertices[3 * v0 + 0] : srcVertices[3 * v1 + 0];
+        float32 y_inside = v0_inside ? srcVertices[3 * v0 + 1] : srcVertices[3 * v1 + 1];
+        float32 z_inside = v0_inside ? srcVertices[3 * v0 + 2] : srcVertices[3 * v1 + 2];
+        float32 x_out = v0_inside ? srcVertices[3 * v1 + 0] : srcVertices[3 * v0 + 0];
+        float32 y_out = v0_inside ? srcVertices[3 * v1 + 1] : srcVertices[3 * v0 + 1];
+        float32 z_out = v0_inside ? srcVertices[3 * v1 + 2] : srcVertices[3 * v0 + 2];
+
+        std::string message = fmt::format("Edge {} connects inside vertex ({}, {}, {}) with outside vertex ({}, {}, {}).  This intersects the bounds of ({}, {}, {}) and ({}, {}, {})",
+                                          std::to_string(i), std::to_string(x_inside), std::to_string(y_inside), std::to_string(z_inside), std::to_string(x_out), std::to_string(y_out),
+                                          std::to_string(z_out), boundingBox[0], boundingBox[1], boundingBox[2], boundingBox[3], boundingBox[4], boundingBox[5]);
+        return MakeErrorResult(to_underlying(ErrorCodes::OutsideVertexError), message);
+      }
+
+      if(behavior == BoundaryIntersectionBehavior::InterpolateOutsideVertex)
```

Comment:
```suggestion
     else if(behavior == BoundaryIntersectionBehavior::InterpolateOutsideVertex)
```
Making this and if/else allows the compiler to either skip second check or potentially treat this as a switch either way will improve preformance